### PR TITLE
Add two more functions for company

### PR DESCRIPTION
**Pull request by jappeace**

### CODE REVIEW

**Comment by psibi** [comment]
@jappeace A hacky way to do that is using `unsafePerfomIO`. I guess it's okay, since it's for testing. Infact, hedgehog-fakedata uses that: https://github.com/parsonsmatt/hedgehog-fakedata/blob/d342c6eb5aeb9990bb36ede1d1f08becc7d71e16/src/Hedgehog/Gen/Faker.hs#L22

**Comment by psibi** [comment]
@jappeace I'm planning to cut a new release in a few days. Let me know if this MR is ready. Thanks!

**Comment by jappeace** [comment]
Hi, I shall see what I can do, I can definitely work on this next Wednesday.

**Comment by jappeace** [comment]
@psibi I added the tests! Please review again when you have time.

**Comment by psibi** [comment]
@jappeace It seems the tests are failing for some GHC versions. Can you see if you can fix them ? Also, I think you need to address some of the earlier comments.

> Sorry for taking so long, I'm a bit overcommitted.

No hurries!

File: src/Faker/Company.hs [hunk 1]
```diff
@@ -46,3 +63,21 @@ $(generateFakeField "company" "type")
...
+domain = do
+  domain <- F.domainSuffix
+  comapanyName <- name
+  pure $ fixupName comapanyName <> "." <> domain
+
+email :: Fake Text
+email = do
+  humanName <-  F.name
+  number <- F.fromRange @Int (0, 999_999_999) -- reasonable uniqueness
+  domain <- domain
```
[the remark this answers is not in the document]
Comment (jappeace) [reply src/Faker/Company.hs]: This was resolved before btw.

File: src/Faker/Company.hs [hunk 2]
```diff
@@ -46,3 +63,21 @@ $(generateFakeField "company" "type")
 -- | SIC code for classifying industries.
 -- @since 0.2.0
 $(generateFakeField "company" "sic_code")
+
+domain :: Fake Text
+domain = do
+  domain <- F.domainSuffix
+  comapanyName <- name
```
[the remark this answers is not in the document]
Comment (jappeace) [reply src/Faker/Company.hs]: this was fixed as well

File: src/Faker/Company.hs [hunk 3]
```diff
@@ -46,3 +63,21 @@ $(generateFakeField "company" "type")
...
+email = do
+  humanName <-  F.name
+  number <- F.fromRange @Int (0, 999_999_999) -- reasonable uniqueness
+  domain <- domain
+  let numText :: Text
+      numText = pack $ show number
+  pure $ fixupName humanName <> "-" <> numText <> "@" <> domain
+
+fixupName :: Text -> Text
+fixupName = Data.Text.filter (\c -> isAlphaNum c || c == '_') . replace " " "_"
```
[the remark this answers is not in the document]
Comment (jappeace) [reply src/Faker/Company.hs:96]: I added the haddock

File: src/Faker/Company.hs [hunk 4]
```diff
@@ -46,3 +63,21 @@ $(generateFakeField "company" "type")
 -- | SIC code for classifying industries.
 -- @since 0.2.0
 $(generateFakeField "company" "sic_code")
+
+domain :: Fake Text
```
[the remark this answers is not in the document]
Comment (jappeace) [reply src/Faker/Company.hs:72]: I also added since annotation

**Comment by jappeace** [comment]
I've been thinking, that fakeQuickcheck function may be extremely usefull for the main library as well, as it makes any `Fake a` function compatible with QuickCheck. Should we move it to there? It'll make fakedata depend on QuicCheck though (now it's just the test suite).

**Comment by psibi** [comment]
> Should we move it to the main library? It'll make fakedata depend on QuicCheck though, now the test suite just depends on it.

Nah, I don't think that is proper. But I'm not opposed to you creating a separate QuickCheck package. Probably create a separate organization for fakedata packages ? We already have two:

* fakedata
* fakedata-parser

We could also have `fakedata-quickcheck` there.

**Comment by jappeace** [comment]
We can do that as well! I guess not everyone will use fakedata with quicheck per se. There is then a small problem in that the fakedata test suite can't depend on that library because fakedata-quickcheck already depends on fakedata. How can we solve that?

**Comment by psibi** [comment]
@jappeace 

> There is then a small problem in that the fakedata test suite can't depend on that library because fakedata-quickcheck already depends on fakedata. How can we solve that?

I guess I'm not understanding it properly, but why do you think it will be a problem ? The situation as I understand is like this:

* fakedata's test suite depends on quickcheck
* A new package named `fakedata-quickcheck` will depend on `fakedata` and `quickcheck`

I don't see any problem with the above setup. But probably I'm missing something. :-)

**Comment by jappeace** [comment]
@psibi In the test suite we use this [function currently](https://github.com/psibi/fakedata/pull/27/files#diff-122a445292e7967a7df45534750aa4bcR205), if we move it to a seperate package the test suite has to depend on that, and I think that causes a cycle (at least I got that when trying to depend on fakedata-hedgehog).

**Comment by psibi** [comment]
@jappeace Ah, I see what you mean. What do you think about this: We don't remove the copy of it in test suite (and there by not depend on fakedata-quickcheck). That would solve this problem:

```
For riskbook alone, this means around 20 functions around custom data types suddenly become generators. 
```

and any other fake functions too...

**Comment by jappeace** [comment]
Sounds good to me, the function is small anyway.

**Comment by jappeace** [comment]
@psibi do you have any more changes you wish from this PR?

**Comment by psibi** [comment]
@jappeace Thanks! Let me know if you intend to create `fakedata-quickcheck`, I would create an organisation and invite you there.

**Comment by jappeace** [comment]
@psibi 
I added another repo which builds here: https://github.com/jappeace/fakedata-quickcheck

If you create the organisation we can move it there.

**Comment by psibi** [comment]
@jappeace I have invited you to the organization. Let me know if you haven't received, I will send a re-invite. Thanks!

**Pull request merged**